internal/temporalcommonv1/payload_json.go

```diff
@@ -22,205 +22,587 @@
 
 package common
 
-// "bytes"
-// "encoding/json"
+import (
+	//"bytes"
 
-// jsonpb "go.temporal.io/api/internal/temporaljsonpb"
-// gojsonpb "google.golang.org/protobuf/encoding/protojson"
+	"encoding"
+	"encoding/base64"
+	gojson "encoding/json"
+	"fmt"
+	"reflect"
+	"sort"
+	"strconv"
+	"strings"
+
+	"go.temporal.io/api/internal/protojson/json"
+	"go.temporal.io/api/temporalproto"
+	//	"google.golang.org/protobuf/encoding/protojson"
+)
+
+const (
+	payloadMetadataKey      = "metadata"
+	payloadDataKey          = "data"
+	shorthandMessageTypeKey = "_protoMessageType"
+)
+
+var logfunc func(msg string, args ...any) (int, error) = fmt.Printf
+
+func SetLogfunc(f func(msg string, args ...any) (int, error)) {
```

Comment:
Should not add package-level function to our public API for this purpose, and definitely should default it to printf, and even if you had to have it, user-facing API needs Godoc

Reply:
Whoops, I meant to delete this. This was for testing specifically